### ISSUE TITLE
Make plugins update respect enable_auto_update

### ISSUE THREAD

**Issue by mathieubouchard** [piwik/piwik#9751]
There's no validation for the value of enable_auto_update before checking if there are plugin updates, which is triggering timeouts when accessing the admin web pages.

If enable_auto_update is set to 0, there should have no check for :
- Piwik update (OK)
- Plugins update (plugins/CorePluginsAdmin/MarketplaceApiClient.php)
- Spam user blacklist update (./plugins/CoreAdminHome/Tasks.php)
- GeoIP update (./plugins/UserCountry/GeoIPAutoUpdater.php)
- etc. ?

**Comment by tsteur** [comment]
Good point, I presume we should respect that setting in the Marketplace. I will see what is possible probably today.

FYI: So far the workflow would be to disable the Marketplace for this by setting `[General]enable_marketplace=0`. 
We will soon introduce a new setting `multi_server_environment = 0/1` which then will disable some updates as well. More about this will follow soon.

**Comment by hpvd** [comment]
hmm I'm not that happy making it global.
Update of data like geo-IP, Spam ... should be handeled separately.

What do you think of making 2 switches,
- one for all software (Piwik itself and all plugins) and
- one for all kind of data (geio, spam etc)

the reason:
- software could introduce new bugs incompatibilities which one may want check manually
- simple data updates shouldn't break anything but only make things better

what do you think?

**Comment by tsteur** [comment]
I won't make it for geoi and spam for now :)

**Comment by hpvd** [comment]
good to hear!

**Comment by mattab** [comment]
Notes:

 * `enable_auto_update` - is currently wrongly implemented. it disables the "Check for new versions" when it should only disable the "Auto update" feature
 * we would need a new setting `enable_check_latest` which would let users disable the actual check for latest version (for core and for marketplace plugins).

**Comment by tsteur** [comment]
Actually, I will remove this issue out of the current milestone again. The setting `enable_auto_update` should actually disable all automatic updates, or at least core and plugins. 

What I will do as part of another feature is to show a "download plugin" button instead of an "update plugin" or "install plugin" if `enable_auto_update` is disabled. However, it will still ping the Marketplace so far. For this we will need another setting like suggested `enable_check_latest` or `enable_outgoing_requests_to_piwik` or something.

**Comment by mattab** [comment]
> For this we will need another setting like suggested enable_check_latest or enable_outgoing_requests_to_piwik or something.

FYI `enable_outgoing_requests_to_piwik ` is covered in https://github.com/piwik/piwik/issues/6324

**Comment by mathieubouchard** [comment]
I added a list of outgoing connections to #6324
